crates/sp-domains-fraud-proof/src/lib.rs

```diff
@@ -31,12 +31,30 @@ pub use runtime_interface::fraud_proof_runtime_interface;
 #[cfg(feature = "std")]
 pub use runtime_interface::fraud_proof_runtime_interface::HostFunctions;
 use sp_api::scale_info::TypeInfo;
+use sp_runtime_interface::pass_by;
+use sp_runtime_interface::pass_by::PassBy;
 use sp_std::vec::Vec;
 use subspace_core_primitives::Randomness;
 
 /// Type to hold required information to verify invalid domain extrinsics root.
 #[derive(Debug, Decode, Encode, TypeInfo, PartialEq, Eq, Clone)]
-pub struct InvalidDomainExtrinsicRootInfo {
+pub struct InvalidDomainExtrinsicRootVerificationInfo {
     pub block_randomness: Randomness,
     pub timestamp_extrinsic: Vec<u8>,
 }
+
+/// Request type to fetch required verification information for fraud proof through Host function.
+#[derive(Debug, Decode, Encode, TypeInfo, PartialEq, Eq, Clone)]
+pub enum FraudProofVerificationInfoRequest {
+    InvalidDomainExtrinsicRoot,
```

Comment:
Please add docs for all enum variants on newly introduced data structures. And the name is odd, are you sure this is a request for invalid domain extrinsic root? Or do we request valid domain extrinsic root for the purposes of verifying an invalid one.

And for these data structures in `sp-*` it would be preferable to specify SCALE codec indices explicitly.

Reply:
> Hm, I think the variant and its comment should indicate what information you're requesting from the client, not what fraud proof verification will use it. You may have multiple fraud proofs requesting the same data.

Updated the comment specific to the variant. I dont think any other fraud proof type would require this info apart from the Fraud proof type. Do you disagree ?

Reply:
That is not the point, it was just a possibility. It is confusing if you don't call things what they are. From last update and comment it seems that maybe it should have already been two variants: for randomness and timestamp.

Reply:
That would be two host functions while this could simply with one call that returns the both and are used at the same time. Sure I agree that it makes the host calls more generic but returning these info bundled together would be nice though I may not be thinking much further ahead and I dont have strong opinion on this. If you think this makes it easier to understand and follow the code path, I willing to change though even though I wouldn't like it much :)

Reply:
I can easily imagine randomness to be useful for some other fraud proofs for example. And going forward for other fraud proofs there might be overlapping pieces of information necessary. The approach where enum variants are saying what they request seems to scale more gracefully than variants that are purpose built for one fraud proof, even though it might be a bit more convenient to handle.

@NingLin-P @ParthDesai do you have an opinion on this?

Reply:
Folks, I would really appreciate if we could move forward and make a decision on this so that we can actually start building the rest of the fraud proofs using the host function. I dont mind taking the route @nazar-pc suggested at this point but we really needs to get feedback and move on. @NingLin-P @ParthDesai

Reply:
I would also prefer  @nazar-pc 's approach since there are overlapping data (i.e. domain runtime code, bundle) required by multiple fraud proof.

Besides we will need the host function to not only fetch data but also actually perform some of the verification steps (where domain runtime API is involved) do we also make them as a variant here?

Reply:
> Besides we will need the host function to not only fetch data but also actually perform some of the verification steps (where domain runtime API is involved) do we also make them as a variant here?

I think so, everything that runtime can't do will go as request to host and back as response, we just need to ensure we are pricing it correctly.